From 820389de68ed0d3aa48b20696c1d730b42262aff Mon Sep 17 00:00:00 2001
From: David Vrabel <david.vrabel@citrix.com>
Date: Wed, 3 Aug 2016 14:53:31 +0100
Subject: [PATCH 2/2] libxencall: release the buffer cache before closing the
 fd

Freeing a buffer may now call the IOCTL_PRIVCMD_HCALL_BUF_UNLOCK ioctl
on the fd, so the fd should only be closed after the buffer cache is
released.

Signed-off-by: David Vrabel <david.vrabel@citrix.com>
---
 tools/libs/call/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/libs/call/core.c b/tools/libs/call/core.c
index 5ca0372..578815a 100644
--- a/tools/libs/call/core.c
+++ b/tools/libs/call/core.c
@@ -65,8 +65,8 @@ int xencall_close(xencall_handle *xcall)
     if ( !xcall )
         return 0;
 
-    rc = osdep_xencall_close(xcall);
     buffer_release_cache(xcall);
+    rc = osdep_xencall_close(xcall);
     xtl_logger_destroy(xcall->logger_tofree);
     free(xcall);
     return rc;
-- 
2.1.4

